packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -4819,4 +4843,37 @@ void main() {
     final EditableText rtlWidget = tester.widget(find.byType(EditableText));
     expect(rtlWidget.textDirection, TextDirection.rtl);
   });
+
+  testWidgets('clipBehavior has expected defaults', (WidgetTester tester) async {
+    await tester.pumpWidget(
+        const CupertinoApp(
+          home: CupertinoTextField(
+          ),
+        ),
+    );
+
+    final CupertinoTextField textField = tester.firstWidget(find.byType(CupertinoTextField));
+    expect(textField.clipBehavior, Clip.hardEdge);
+  });
+
+  testWidgets('Overflow clipBehavior none golden', (WidgetTester tester) async {
+    final Widget widget = CupertinoApp(
+      home: RepaintBoundary(
+        key: const ValueKey<int>(1),
+        child: CupertinoTextField(
+          controller: OverflowWidgetTextEditingController(),
+          clipBehavior: Clip.none,
+        ),
+      ),
+    );
+    await tester.pumpWidget(widget);
+
+    final CupertinoTextField textField = tester.firstWidget(find.byType(CupertinoTextField));
+    expect(textField.clipBehavior, Clip.none);
```

Comment:
Nit: Maybe also find the EditableText and expect that it is `Clip.none` too?  Here and below.